New SKL==1.0 update causes mahalanobis to break

With  `scikit_learn==1.0`, we get:

```
        if metric == "mahalanobis" and "VI" not in kwds:
            if X is Y:
                VI = np.linalg.inv(np.cov(X.T)).T
            else:
>               raise ValueError(
                    "The 'VI' parameter is required for the mahalanobis metric "
                    "when Y is passed."
                )
E               ValueError: The 'VI' parameter is required for the mahalanobis metric when Y is passed.
```

See [run](https://github.com/microsoft/hummingbird/pull/543/checks?check_run_id=3723686368).

Last known good [run](https://github.com/microsoft/hummingbird/runs/3690424401) four days ago had `scikit_learn-0.24.2`.

--- a/tests/test_sklearn_kneighbors.py
+++ b/tests/test_sklearn_kneighbors.py
@@ -33,6 +33,7 @@ class TestSklearnKNeighbors(unittest.TestCase):
             elif metric == "mahalanobis":
                 V = np.cov(X.T)
                 metric_params["V"] = V
+                metric_params["VI"] = np.linalg.inv(V)
 
             model = KNeighborsClassifier(
                 n_neighbors=n_neighbors, algorithm=algorithm, weights=weights, metric=metric, metric_params=metric_params

